RosNet/Field/FieldValue.cs

```diff
@@ -1,146 +1,104 @@
-                    return 2;
-                case PrimitiveType.Int32:
-                case PrimitiveType.Uint32:
-                case PrimitiveType.Float32:
-                case PrimitiveType.String:
-                    return 4;
-                case PrimitiveType.Int64:
-                case PrimitiveType.Uint64:
-                case PrimitiveType.Float64:
-                case PrimitiveType.Time:
-                case PrimitiveType.Duration:
-                    return 8;
-                default:
-                    return 0;
-            }
+            case PrimitiveType.BOOL:
+            case PrimitiveType.INT8:
+            case PrimitiveType.UINT8:
+            case PrimitiveType.BYTE:
+            case PrimitiveType.CHAR:
+                return 1;
+            case PrimitiveType.INT16:
+            case PrimitiveType.UINT16:
+                return 2;
+            case PrimitiveType.INT32:
+            case PrimitiveType.UINT32:
+            case PrimitiveType.FLOAT32:
+            case PrimitiveType.STRING:
+                return 4;
+            case PrimitiveType.INT64:
+            case PrimitiveType.UINT64:
+            case PrimitiveType.FLOAT64:
+            case PrimitiveType.TIME:
+            case PrimitiveType.DURATION:
+                return 8;
+            default:
+                return 0;
         }
+    }
 
-        /// <summary>
-        /// Creates string with value if printValue is true, else string without value (used for messageDefinition)
-        /// </summary>
-        /// <returns>String with value if printValue is true, else returns string without value</returns>
-        public virtual string ToString(bool printValue)
+    /// <summary>
+    /// Creates string with value
+    /// </summary>
+    /// <returns>String with value</returns>
+    public override string ToString()
+    {
+        if (this.Value.Length == 0)
         {
-            if (printValue)
-            {
-                return this.ToString(); 
-            }
-            return ($"{DataType} {Name}");
-
+            return ($"{DataType} {Name} noValue");
         }
+        return ($"{DataType} {Name} {PrettyValue()}");
+    }
 
-        /// <summary>
-        /// Creates string with value
-        /// </summary>
-        /// <returns>String with value</returns>
-        public override string ToString()
-        {
-            if (this.Value.Length == 0)
-            {
-                return ($"{DataType} {Name} noValue");
-            }
-            return ($"{DataType} {Name} {PrettyValue()}");
-        }
+    /// <summary>
+    /// Creates string of Value using datatype
+    /// </summary>
+    /// <returns>String of Value</returns>
+    public string PrettyValue() => this.DataType switch
+    {
+        PrimitiveType.BOOL => ($"{BitConverter.ToBoolean(this.Value)}"),
+        PrimitiveType.INT8 => ((int)this.Value.First()).ToString(),
+        PrimitiveType.UINT8 => ((uint)this.Value.First()).ToString(),
```

Comment:
```suggestion
    public string PrettyValue() => (this.DataType switch
    {
        PrimitiveType.BOOL => BitConverter.ToBoolean(this.Value),
        PrimitiveType.INT8 => (int)this.Value.First(),
        PrimitiveType.UINT8 => (uint)this.Value.First(),
        ...,
}).ToString()
```
then you don't have all the `($"{}")`-ing